InMobi/src/main/java/com/mopub/nativeads/InMobiNative.kt

```diff
@@ -0,0 +1,320 @@
+        } catch (e: SdkNotInitializedException) {
+            MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "Error while creating InMobiNative Object. " + e.message)
+            handleNativeInitializationFailure()
+            return
+        }
+        inMobiStaticNativeAd?.setExtras(InMobiAdapterConfiguration.inMobiTPExtras)
+
+        val adMarkup = serverExtras[DataKeys.ADM_KEY]
+        if (adMarkup != null) {
+            MoPubLog.log(
+                AdapterLogEvent.CUSTOM, InMobiBanner.ADAPTER_NAME,
+                "Ad markup for InMobi banner ad request is present. Will make Advanced Bidding ad request " +
+                        "using markup: " + adMarkup
+            )
+            inMobiStaticNativeAd?.loadAd(adMarkup?.toByteArray())
+        } else {
+            MoPubLog.log(
+                AdapterLogEvent.CUSTOM, InMobiBanner.ADAPTER_NAME,
+                "Ad markup for InMobi banner ad request is not present. Will make traditional ad request "
+            )
+            inMobiStaticNativeAd?.loadAd()
+        }
+    }
+
+    private fun handleNativeInitializationFailure() {
+        mNativeListener?.onNativeAdFailed(NativeErrorCode.NATIVE_ADAPTER_CONFIGURATION_ERROR)
+    }
+
+    class InMobiNativeAd internal constructor(
+        private val mContext: Context,
+        private val mCustomEventNativeListener: CustomEventNativeListener,
+        placementId: Long
+    ) : BaseNativeAd() {
+
+        private val mNativeClickHandler: NativeClickHandler = NativeClickHandler(mContext)
+        private val mInMobiNative: InMobiNative
+        private var mIsImpressionRecorded = false
+        private var mIsClickRecorded = false
+
+        private val nativeAdEventListener: NativeAdEventListener = object : NativeAdEventListener() {
+            override fun onAdLoadSucceeded(ad: InMobiNative, info: AdMetaInfo) {
+                MoPubLog.log(AdapterLogEvent.CUSTOM, TAG, "InMobi Native Ad loaded successfully")
+                val imageUrls: MutableList<String> = ArrayList()
+                val iconImageUrl = adIconUrl
+                imageUrls.add(iconImageUrl)
+
+                NativeImageHelper.preCacheImages(mContext, imageUrls, object : NativeImageHelper.ImageListener {
+                    override fun onImagesCached() {
+                        mCustomEventNativeListener.onNativeAdLoaded(this@InMobiNativeAd)
+                    }
+
+                    override fun onImagesFailedToCache(errorCode: NativeErrorCode) {
+                        mCustomEventNativeListener.onNativeAdFailed(errorCode)
+                    }
+                })
+            }
+
+            override fun onAdLoadFailed(
+                inMobiNative: InMobiNative,
+                inMobiAdRequestStatus: InMobiAdRequestStatus
+            ) {
+                super.onAdLoadFailed(inMobiNative, inMobiAdRequestStatus)
+                var errorMessage = "Failed to load Native Strand:"
+                when (inMobiAdRequestStatus.statusCode) {
```

Comment:
There is a common function in `InMobiAdapterConfiguration` class `getMoPubErrorCode()`. 

Use it here as well and add more branches in common function if needed. Also keep the log in similar format as done here https://github.com/InMobi/mopub-android-mediation/blob/master/InMobi/src/main/java/com/mopub/mobileads/InMobiBanner.kt#L97

Reply:
getMoPubErrorCode() returns `MoPubErrorCode` whereas Native Ad uses `NativeErrorCode` . So can't use that in InMobiNative class.
Made log statement uniform.

Reply:
Ah okay